include/SFML/Audio/Sound.hpp

```diff
@@ -244,6 +246,11 @@ class SFML_AUDIO_API Sound : public SoundSource
     ////////////////////////////////////////////////////////////
     struct Impl;
     const std::unique_ptr<Impl> m_impl; //!< Implementation details
+
+    ////////////////////////////////////////////////////////////
+    // Lifetime tracking
+    ////////////////////////////////////////////////////////////
+    SFML_DEFINE_LIFETIME_DEPENDANT(SoundBuffer);
```

Comment:
`sf::Sound` and `sf::SoundBuffer` already perform their own internal lifetime checks where sound buffers are de-registered with the `Sound`s that use them when the `SoundBuffer` is destroyed. Do these new lifetime checks means we can remove all that machinery from `Sound` and `SoundBuffer`?